runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -242,15 +252,33 @@ void doSerializeInternal(ByteBuf buf) {
             if (data == null) {
                 int lengthIndex = buf.writerIndex();
                 buf.writeInt(0);
-                Serializers.CORFU.serialize(payload.get(), buf);
+                if (hasPayloadCodec()) {
```

Comment:
Can you please add some comments explaining these different cases?

Reply:
While comments will make it more understandable, I wonder if it is possible to simplify the control-flow.